src/network.js

```diff
@@ -0,0 +1,38 @@
+import promisify from 'es6-promisify';
+import dns from 'dns';
+import ip from 'ip';
+import log, {WARN} from "./log";
+
+const CACHABLE_NETWORK_ERRORS = [
+	'HPE_INVALID_CONSTANT' // May trigger if eg chipsoft Sharepoint send a content-length of 0 but secretly appends data
+];
+const NETWORK_ERRORS = [].concat(CACHABLE_NETWORK_ERRORS);
```

Comment:
Minor: I'd go with the spread operator, I find that slightly more readable than concat.